### PR TITLE
T13271 (fixup)

### PR DESCRIPTION
**Pull request by tchx84**
https://phabricator.endlessm.com/T13271

### CODE REVIEW

**Comment by tchx84** [comment]
The previous commit fixed the issue with the missing titles, but, introduced yet another regressions:

a) non-markup text gets escaped and shows incorrectly. (this is fixed by the previous commit)
b) markup text which does get escaped, keep changing so the label it set kind-of-recursively (I am still trying to figure out how to fix this).

**Comment by tchx84** [comment]
I am adding an extra test to make sure this widget is robust enough against these confusing ampersand symbols...

File: tests/js/app/widgets/testFormattableLabel.js [hunk 1]
```diff
@@ -55,4 +56,29 @@ describe('Formattable Label', function () {
...
+        win.add(label);
+        win.show_all();
+        label.get_style_context().add_provider(cssProvider,
+                                                Gtk.StyleProvider.PRIORITY_APPLICATION + 10);
+
+        Utils.update_gui();
+        let text = "a & b";
+
+        label.use_markup = true;
+        label.label = GLib.markup_escape_text(text, -1);
```
Comment (mattdangerw) [inline tests/js/app/widgets/testFormattableLabel.js]: Why do you need to call `GLib.markup_escape_text(text, -1);` here? Won't the fact that you have `use_markup = true` plus that custom label setter that calls  `_format_label` mean this gets called already with your changes above?

File: tests/js/app/widgets/testFormattableLabel.js [hunk 2]
```diff
@@ -55,4 +56,29 @@ describe('Formattable Label', function () {
...
+
+        cssProvider.load_from_data ("* { \
+            -EknFormattableLabel-text-transform: 'uppercase';\
+        }");
+
+        let win = new Gtk.OffscreenWindow();
+        win.add(label);
+        win.show_all();
+        label.get_style_context().add_provider(cssProvider,
+                                                Gtk.StyleProvider.PRIORITY_APPLICATION + 10);
```
Comment (mattdangerw) [inline tests/js/app/widgets/testFormattableLabel.js]: nit: remove extra space

File: tests/js/app/widgets/testFormattableLabel.js (hunk 1 above)
Comment (tchx84) [inline tests/js/app/widgets/testFormattableLabel.js]: The main reason is that ```label.label = "a & b"```; is not valid input when use_markup is true (this is also how the base GtkLabel behaves), "&" is treated a special character for escaping symbols e.g. ```"&amp;"```, so text must be escaped. Please keep this in mind for what I explain next.

I know is confusing let me try explain:

1) Inside "_format_label", we use "parse_markup" and this "reverts" the escaped symbols, so

e.g. ```"a &amp; b"``` turns back into ```"a & b"```;
If we pass, a plain "&" the parser will reject it (as I explained, ```&`` is not valid input with use_markup true).

2) then, we do the transformations.

e.g. ```"a & b"``` turns into ```"A & B"```

3) then we set the text back to widget, using this.set_label, BUT we need to escape these symbols back or it won't work:

e.g. if use_markup is true, then we will try to ```set_label("A & B")```, which (as I explained before) is not valid, must be escaped back so we do ```set_label("A &amp; B")```.

Let me know If this explanation is not clear. But the point is, we do need to escape text before setting it to the label (when use_markup is true).

The logic is completely different if use_markup is false, that is why need all these ```if (this.use_markup)``` checks.

**Comment by tchx84** [comment]
@mattdangerw fixed the nit!

**Comment by ptomato** [comment]
Found some nitpicks, but due to the deadline I am fixing them myself. **CR pass**

**Pull request merged**